x-pack/plugin/core/src/main/resources/ecs-dynamic-mappings.json

```diff
@@ -0,0 +1,195 @@
+{
+  "template": {
+    "mappings": {
+      "dynamic_templates": [
+        {
+          "ecs_timestamp": {
+            "mapping": {
+              "ignore_malformed": false,
+              "type": "date"
+            },
+            "match": "@timestamp"
+          }
+        },
+        {
+          "ecs_message_match_only_text": {
+            "mapping": {
+              "type": "match_only_text"
+            },
+            "match": "message"
+          }
+        },
+        {
+          "ecs_ip": {
+            "mapping": {
+              "type": "ip"
+            },
+            "match": [
+              "ip",
+              "*_ip"
+            ],
+            "match_mapping_type": "string"
+          }
+        },
+        {
+          "ecs_wildcard": {
+            "mapping": {
+              "type": "wildcard"
+            },
+            "path_match": [
+              "*.io.text",
+              "registry.data.strings",
+              "*url.path"
+            ]
+          }
+        },
+        {
+          "ecs_path_match_wildcard_and_match_only_text": {
+            "mapping": {
+              "fields": {
+                "text": {
+                  "type": "match_only_text"
+                }
+              },
+              "type": "wildcard"
+            },
+            "path_match": [
+              "*.body.content",
+              "*url.full",
+              "*url.original"
+            ]
+          }
+        },
+        {
+          "ecs_match_wildcard_and_match_only_text": {
+            "mapping": {
+              "fields": {
+                "text": {
+                  "type": "match_only_text"
+                }
+              },
+              "type": "wildcard"
+            },
+            "match": [
+              "command_line",
+              "stack_trace"
+            ]
+          }
+        },
+        {
+          "ecs_path_match_keyword_and_match_only_text": {
+            "mapping": {
+              "fields": {
+                "text": {
+                  "type": "match_only_text"
+                }
+              },
+              "type": "keyword"
+            },
+            "path_match": [
+              "*file.path",
+              "*file.target_path",
+              "*os.full",
+              "user_agent.original"
+            ]
+          }
+        },
+        {
+          "ecs_match_keyword_and_match_only_text": {
+            "mapping": {
+              "fields": {
+                "text": {
+                  "type": "match_only_text"
+                }
+              },
+              "type": "keyword"
+            },
+            "match": [
+              "title",
```

Comment:
Is this matching flattened and non flattened documents?

`{ "foo.title": "test"}`

`{ "foo": { "title" : "test" } } `

I assume it does, but want to double check as we had recently quite a few discussions around different behaviours on flat vs object.

Reply:
My limited understanding is that it actually shouldn't....
Once `foo` is mapped to `flattened` doesn't this mean that its subfields are not mapped?

Reply:
I didn't mean by the word `flattened` that the mapping would be `flattened`, just the way the document is sent to the _bulk endpoint.

Reply:
Seems to be properly matching as expected - I added such [flattened entry](https://github.com/elastic/elasticsearch/pull/96171/commits/8956c7f3f9b0e177b014c4d5e8ab5abd47861541#diff-c5877225937baee0eb47d89e6502503d5813b1df3048a6a8e1c48c0f2a6a2855R238) (that is itself nested within an object) to the test, as well as an [assertion](https://github.com/elastic/elasticsearch/pull/96171/commits/8956c7f3f9b0e177b014c4d5e8ab5abd47861541#diff-c5877225937baee0eb47d89e6502503d5813b1df3048a6a8e1c48c0f2a6a2855R330) that it is mapped as expected.